src/Microsoft.ML.Onnx/OnnxUtils.cs

```diff
@@ -295,56 +295,40 @@ public ModelArgs(string name, TensorProto.Types.DataType dataType, List<long> di
             Contracts.CheckNonEmpty(colName, nameof(colName));
 
             TensorProto.Types.DataType dataType = TensorProto.Types.DataType.Undefined;
-            DataKind rawKind;
+            Type rawType;
             if (type is VectorType vectorType)
-                rawKind = vectorType.ItemType.RawKind;
-            else if (type is KeyType keyType)
-                rawKind = keyType.RawKind;
+                rawType = vectorType.ItemType.RawType;
+            else
+                rawType = type.RawType;
+
+            if (rawType == typeof(bool))
+                dataType = TensorProto.Types.DataType.Float;
+            else if (rawType == typeof(ReadOnlyMemory<char>))
+                dataType = TensorProto.Types.DataType.String;
+            else if (rawType == typeof(sbyte))
+                dataType = TensorProto.Types.DataType.Int8;
+            else if (rawType == typeof(byte))
+                dataType = TensorProto.Types.DataType.Uint8;
+            else if (rawType == typeof(short))
+                dataType = TensorProto.Types.DataType.Int16;
+            else if (rawType == typeof(ushort))
+                dataType = TensorProto.Types.DataType.Uint16;
+            else if (rawType == typeof(int))
+                dataType = TensorProto.Types.DataType.Int32;
+            else if (rawType == typeof(uint))
+                dataType = TensorProto.Types.DataType.Int64;
```

Comment:
@wschin - this one looks incorrect as well.

Reply:
It appears both BL and U4 mapped these values when this repo was created - [https://github.com/dotnet/machinelearning/blob/f0e639af5ffdc839aae8e65d19b5a9a1f0db634a/src/Microsoft.ML.Data/Model/Onnx/OnnxUtils.cs#L296-L328](https://github.com/dotnet/machinelearning/blob/f0e639af5ffdc839aae8e65d19b5a9a1f0db634a/src/Microsoft.ML.Data/Model/Onnx/OnnxUtils.cs#L296-L328)

---
In reply to: [248108591](https://github.com/dotnet/machinelearning/pull/2143#discussion_r248108591) [](ancestors = 248108591)

Reply:
also /cc @zeahmed and @jignparm - in case they understand both of these type mappings.

Reply:
Since we always translate uint to ONNX Int64, this should be fine.

---
In reply to: [248112875](https://github.com/dotnet/machinelearning/pull/2143#discussion_r248112875) [](ancestors = 248112875)